src/applications/simple-forms/21-0966/config/submit-transformer.js

```diff
@@ -4,21 +4,16 @@ export default function transformForSubmit(formConfig, form) {
   const transformedData = JSON.parse(
     sharedTransformForSubmit(formConfig, form),
   );
-  const {
-    benefitSelectionCompensation,
-    benefitSelectionPension,
-  } = transformedData;
 
-  if (benefitSelectionCompensation) {
-    transformedData.benefitSelection = {
-      compensation: true,
-    };
-  }
-  if (benefitSelectionPension) {
-    transformedData.benefitSelection = {
-      pension: true,
-    };
-  }
-
-  return JSON.stringify(transformedData);
+  return JSON.stringify({
+    ...transformedData,
+    benefitSelection: {
+      compensation:
+        transformedData.benefitSelection?.compensation ||
+        !!transformedData.benefitSelectionCompensation,
+      pension:
+        transformedData.benefitSelection?.pension ||
+        !!transformedData.benefitSelectionPension,
```

Comment:
Why do we need the `!!` lines? Shouldn't `transformedData.benefitSelection?.compensation` be sufficient?

Reply:
That's a good point, I can't remember why I would've done that for the independent benefit data and not the group checkbox one...I'll revisit this in another PR since this one is passing all checks on this commit.